core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryActivity.kt

```diff
@@ -0,0 +1,141 @@
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.Filter
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.OnItemLongClick
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.OnSwitch
+import org.kiwix.kiwixmobile.core.history.viewmodel.HistoryViewModel
+import org.kiwix.kiwixmobile.core.history.viewmodel.State
+import org.kiwix.kiwixmobile.core.history.viewmodel.State.NoResults
+import org.kiwix.kiwixmobile.core.history.viewmodel.State.Results
+import org.kiwix.kiwixmobile.core.history.viewmodel.State.SelectionResults
+import org.kiwix.kiwixmobile.core.utils.SimpleTextListener
+import java.util.ArrayList
+import javax.inject.Inject
+
+class HistoryActivity : OnItemClickListener, BaseActivity() {
+
+  val activityComponent by lazy { coreActivityComponent }
+  private val historyList: List<HistoryListItem> = ArrayList()
+  private val fullHistory: List<HistoryListItem> = ArrayList()
+  val search = menu_history_search
+  private val deleteList: List<HistoryListItem> = ArrayList()
+  @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+  private lateinit var recyclerView: RecyclerView
+  private lateinit var historySwitch: Switch
+  private val historyViewModel by lazy { viewModel<HistoryViewModel>(viewModelFactory) }
+  private val compositeDisposable = CompositeDisposable()
+
+  private val historyAdapter: HistoryAdapter2 by lazy {
+    HistoryAdapter2(
+      HistoryItemDelegate(deleteList, this)
+    )
+  }
+
+  override fun injection(coreComponent: CoreComponent) {
+    coreComponent.activityComponentBuilder()
+      .activity(this)
+      .build()
+      .inject(this)
+    activityComponent.inject(this);
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_history)
+    setSupportActionBar(toolbar)
+    val actionBar = supportActionBar
+    actionBar?.setDisplayHomeAsUpEnabled(true)
+    actionBar?.setTitle(string.history)
+    recyclerView = recycler_view
+    recyclerView.adapter = historyAdapter
+    recyclerView.layoutManager = LinearLayoutManager(this, RecyclerView.VERTICAL, false)
+
+    historySwitch = history_switch
+    historySwitch.setOnCheckedChangeListener { button, isChecked ->
+      historyViewModel.actions.offer(OnSwitch(isChecked))
+    }
+    historySwitch.isChecked = !sharedPreferenceUtil.showHistoryCurrentBook
+  }
+
+  override fun onDestroy() {
+    compositeDisposable.clear()
+    super.onDestroy()
+  }
+
+  override fun onCreateOptionsMenu(menu: Menu): Boolean {
+    menuInflater.inflate(R.menu.menu_history, menu)
+    val search = menu.findItem(id.menu_history_search).actionView as SearchView
+    search.queryHint = getString(string.search_history)
+    search.setOnQueryTextListener(SimpleTextListener {
+      historyViewModel.actions.offer(Filter(it))
+    })
+    historyViewModel.state.observe(this, Observer(::render))
```

Comment:
the reason you aren't triggering any effects is you have to also subscribe to the `sideEffects` stream provided in the ViewModel